Code/Editor/Plugins/ProjectSettingsTool/ProjectSettingsContainer.cpp

```diff
@@ -282,35 +311,49 @@ namespace ProjectSettingsTool
         return m_projectJson.m_document->GetAllocator();
     }
 
-    const char* ProjectSettingsContainer::GetFailedLoadingPlistText()
-    {   
-        return "Failed to load info.plist"; 
+    void ProjectSettingsContainer::LoadJson(JsonSettings& jsonSettings)
+    {
+        StringOutcome outcome = ReadConfigFile(jsonSettings.m_path, jsonSettings.m_rawData);
+        if (!outcome.IsSuccess())
+        {
+            m_errors.push(SettingsError(AZStd::string::format("Failed to load %s", jsonSettings.m_path.c_str()), outcome.GetError(), true /*shouldAbort*/));
+        }
+
+        jsonSettings.m_document->Parse(jsonSettings.m_rawData.c_str());
```

Comment:
Curious why the document is parsed even if there were errors loading/reading the file.  Seems the result would be the same (empty rawData, empty document), so why not just return on an error?

Reply:
It's just how this class has been designed. At the moment all the loads are called from the constructor, so there is no point in returning an error, it records them into `m_error` so later in `ProjectSettingsToolWindow::IfErrorShowThenExit()` collects them and handles them.

I think `ProjectSettingsToolWindow` and `ProjectSettingsContainer` overall can be improved. But the point of this PR is just to make it work without changing its architecture or its API design, that'd be too much of a big change for just getting android settings to work.

But I would recommend revisit this plugin and revamp it in the future.